tests/src/python/test_qgsrelationpostgres.py

```diff
@@ -0,0 +1,63 @@
+# -*- coding: utf-8 -*-
+"""QGIS Unit tests for relations with postgresql provider.
+
+.. note:: This program is free software; you can redistribute it and/or modify
+it under the terms of the GNU General Public License as published by
+the Free Software Foundation; either version 2 of the License, or
+(at your option) any later version.
+"""
+__author__ = 'Loïc Bartoletti'
+__date__ = '27/07/2020'
+__copyright__ = 'Copyright 2020, The QGIS Project'
+
+import qgis  # NOQA
+
+import os
+
+from qgis.core import (
+    QgsVectorLayer,
+    QgsProject,
+    QgsRelationManager
+)
+from qgis.gui import (
+    QgsMapCanvas
+)
+from qgis.testing import start_app, unittest
+
+start_app()
+
+
+class TestQgsRelationPostgresql(unittest.TestCase):
+
+    @classmethod
+    def setUpClass(cls):
+        """
+        Setup the involved layers and relations for a n:m relation
+        :return:
+        """
+	cls.mapCanvas = QgsMapCanvas()
+        cls.dbconn = 'service=\'qgis_test\''
+        if 'QGIS_PGTEST_DB' in os.environ:
+            cls.dbconn = os.environ['QGIS_PGTEST_DB']
+        # Create test layer
+	tables = [ 'c_amgmt_amgmt_lot', 'c_batiment_bat_lot', 'c_ens_immo_amgmt', 'c_ens_immo_bat', 'c_terrain_ens_immo', 't_actes', 't_adresse', 't_amgmt', 't_amgmt_lot', 't_bat', 't_bat_lot', 't_ens_immo', 't_terrain' ]
+	vl_tables = ['vl_c_amgmt_amgmt_lot','vl_c_batiment_bat_lot','vl_c_ens_immo_amgmt','vl_c_ens_immo_bat','vl_c_terrain_ens_immo','vl_t_actes','vl_t_adresse','vl_t_amgmt','vl_t_amgmt_lot','vl_t_bat','vl_t_bat_lot','vl_t_ens_immo','vl_t_terrain' ]
+
+	for i in range(len(tables)):
```

Comment:
Does this still belong to `setUpClass`? I think the indentation is not properly aligned.